### PR TITLE
Add OSG 23 images (SOFTWARE-5673)

### PR DESCRIPTION
**Pull request by brianhlin**
I left all the images that directly referenced EL8 as is since I didn't want to force them back to EL7 for OSG 3.6 images

### CODE REVIEW

**Comment by brianhlin** [comment]
Things are looking pretty solid except for `ospool-display`, `gracc-apel`, and `vo-frontend`:

@djw8605 OSG 23 EL9 builds for `gracc-apel` are failing because of https://github.com/opensciencegrid/images/actions/runs/6267522494/job/17020771998?pr=149#step:2:911. Any chance we can get EL9 argo RPMs?

@rynge fifemon is blowing up on the install for OSG 3.6 EL7. Any ideas on that? Otherwise for EL9 there's this really strange conflict between `curl` and `curl-minimal` that I don't quite understand.

`vo-frontend` looks like it's failing because of missing python3-scitokens. Derek said that this should have gotten into EPEL9 last night so hopefully this one just goes away

**Comment by rynge** [comment]
So exciting! Are we just leaving `BASE_OSG_SERIES=3.6` as default for now?

We are in the process of removing fifemon - I'm just waiting for the last final ok from ToddT on this. I would not invest any time on that image.

**Comment by jthiltges** [comment]
> @djw8605 OSG 23 EL9 builds for `gracc-apel` are failing because of https://github.com/opensciencegrid/images/actions/runs/6267522494/job/17020771998?pr=149#step:2:911. Any chance we can get EL9 argo RPMs?

EL8 support is in progress for apel-ssm: https://github.com/apel/ssm/issues/123

**Comment by brianhlin** [comment]
> So exciting! Are we just leaving BASE_OSG_SERIES=3.6 as default for now?

These are really just for dev convenience so that you don't have to pass args when building locally. The GHA drives all the images that are actually produced (3.6, el7 and 23, el9 by default)

> EL8 support is in progress for apel-ssm: https://github.com/apel/ssm/issues/123

What are the chances for EL9 support, to? 😄

**Comment by brianhlin** [comment]
@mwestphall this is ready for review. The `ospool-display` failures are expected (see discussion above) and shouldn't affect pushes of other images

**Review by mwestphall — APPROVED** [review]
LGTM

File: .github/workflows/build-containers.yml [hunk 1]
```diff
@@ -108,69 +120,80 @@ jobs:
...
+          base_os: ${{ matrix.series.os }}
           registry_url: hub.opensciencegrid.org
           registry_user: ${{ secrets.OSG_HARBOR_ROBOT_USER }}
           registry_pass: ${{ secrets.OSG_HARBOR_ROBOT_PASSWORD }}
 
-      - name: Push to Harbor (3.6-testing)
-        if: always()
-        uses: opensciencegrid/push-container-action@v0.5.0
+      - name: Push to Harbor (${{ matrix.series.name }}-testing)
+        if: matrix.series.name != '23'
```
Comment (mwestphall) [inline .github/workflows/build-containers.yml:142]:
```suggestion
        # FIXME: reenable for OSG 23 builds when available
        if: matrix.series.name != '23'
```

File: .github/workflows/build-containers.yml [hunk 2]
```diff
@@ -108,69 +120,80 @@ jobs:
...
+          base_os: ${{ matrix.series.os }}
           registry_url: docker.io
           registry_user: ${{ secrets.DOCKER_USERNAME }}
           registry_pass: ${{ secrets.DOCKER_PASSWORD }}
 
-      - name: Push to Docker Hub (3.6-testing)
-        if: always()
-        uses: opensciencegrid/push-container-action@v0.5.0
+      - name: Push to Docker Hub (${{ matrix.series.name }}-testing)
+        if: matrix.series.name != '23'
```
Comment (brianhlin) [inline .github/workflows/build-containers.yml:178]:
```suggestion
        # FIXME: reenable for OSG 23 builds when available
        if: matrix.series.name != '23'
```

**Pull request merged**